pkg/resource/training_job/sdk.go

```diff
@@ -1060,6 +1060,9 @@ func (rm *resourceManager) sdkUpdate(
 	if warmpool_diff && profiler_diff {
 		return latest, errors.New("[ACK_SM] Cannot update Warm pool and Profiler at the same time.")
 	}
+	if !warmpool_diff && !profiler_diff {
+		return latest, errors.New("[ACK_SM] Only Warm Pool or Profiler can be updated")
```

Comment:
>Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not fmt.Errorf("Something bad"), so that log.Printf("Reading %s: %v", filename, err) formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.

from https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Also the `[ACK_SM]` is just adding noise to the logs. Users already see this information in the controller logs, along with reconciller name, namespace and a lot of other stuff.

Reply:
Good point, I will use lowercase errors.

> Also the [ACK_SM] is just adding noise to the logs.

[ACK_SM] is used as a terminal prefix in UpdateConditions(https://github.com/aws-controllers-k8s/sagemaker-controller/pull/179/files#diff-06bfc8acec53697c68f4ed8d7e1f0192ec628506edeb9a769036611a78dc2ae1). The alternative I considered manually iterating through all possible terminal errors(which isn't maintainable, might cause errors if the sagemaker server errors change). I will rename that prefix to something like acksm_terminal: as its a bit more clear.

Reply:
Rather than using matching for this, we already have a `TerminalError` type, which the common runtime detects and applies as a terminal condition - https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/errors/error.go#L104-L106